### ISSUE TITLE
PT004 should operate differently for fixtures with autouse=True

### ISSUE THREAD

**Issue by djmattyg007** [m-burst/flake8-pytest-style#141]
If a fixture has been created with `autouse=True`, it likely doesn't return a value, and it's also something I should be manually specifying for any of my tests. Instead of requiring me to change this fixture names to start with an underscore, I feel like it would be better for there to be a separate rule that checks to make sure I don't manually use an autouse fixture.

**Comment by m-burst** [comment]
Hi @djmattyg007,

I'm not sure I agree with this.
In my experience I have seen and written a bunch of tests with the following pattern:

```py
@pytest.fixture(autouse=True)
def admin_user():
    return create_admin_user_in_db()

def test_foo():
    # test something that requires admin_user to exist in db
    ...

def test_bar(admin_user):
    # do something with admin_user
    ...
```

So in my opinion an autouse fixture, as any other fixture, may or may not return a value, and if it does, it can be OK to manually use an autouse fixture.

**Comment by djmattyg007** [comment]
Fair enough. How would you feel about creating a new rule (that doesn't have to be enabled by default) to warn if an autouse fixture is manually required somewhere, like in your example? I'd be happy to take a look at implementing this, though I might need some help.
